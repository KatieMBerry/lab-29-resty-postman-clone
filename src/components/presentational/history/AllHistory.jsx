import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';

const AllHistory = ({ historyList }) => {
    const historyData = historyList.map(listItem => (
        <li className="historyListItem"
            key={listItem.id}>
            <HistoryItem {...listItem} />
        </li>
    ));

    return (
        <ul data-testid="historyList">
            {historyData}
        </ul>
    );
}

AllHistory.propTypes = {
    historyList: PropTypes.arrayOf(
        PropTypes.shape({
            requestMethod: PropTypes.string.isRequired,
            urlInput: PropTypes.string.isRequired
        })
    )
}

export default HistoryItem;
