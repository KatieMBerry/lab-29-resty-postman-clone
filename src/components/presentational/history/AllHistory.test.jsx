import React from 'react';
import { render, cleanup } from '@testing-library/react';
import AllHistory from './AllHistory';

describe('AllHistory component', () => {
    afterEach(() => cleanup());
    it('renders AllHistory', () => {
        const { asFragment } = render(<AllHistory />);
        expect(asFragment()).toMatchSnapshot();
    });
});
