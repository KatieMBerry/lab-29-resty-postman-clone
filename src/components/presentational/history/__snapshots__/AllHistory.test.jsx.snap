// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllHistory component renders AllHistory 1`] = `
<DocumentFragment>
  <div>
    <h6 />
    <p />
  </div>
</DocumentFragment>
`;
